packages/cli/src/cmds/cmd_build/compile.rs

```diff
@@ -1,28 +1,159 @@
-// This modules handles compiling c/c++/asm/rust code
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025 Megaton contributors
 
-use cu::Result;
+use std::{
+    path::{Path, PathBuf},
+    str::FromStr,
+    time::SystemTime,
+};
+
+use cu::pre::*;
 
 use super::{Flags, Lang, RustCrate, SourceFile};
+use crate::{cmds::cmd_build::BuildEnvironment, env::environment};
+
+#[derive(Serialize, Deserialize, PartialEq, Eq)]
+struct CompileCommand {
+    compiler: PathBuf,
+    source: PathBuf,
+    args: Vec<String>,
+}
+
+impl CompileCommand {
+    fn new(compiler_path: &Path, src_file: &Path, out_file: &Path, flags: &Vec<String>) -> Self {
+        let mut argv = flags.clone();
+        argv.push(src_file.to_string_lossy().into_owned());
```

Comment:
let's error if not UTF-8. cu has a util for it https://cu.pistonite.dev/pistonite_cu/trait.PathExtension.html